### ISSUE TITLE
RedisGraph does not currently support line_comment

### ISSUE THREAD

**Issue by odanoburu** [RedisGraph/RedisGraph#1323]
hello, I got the error `RedisGraph does not currently support line_comment` when attempting to port an application from Neo4j to RedisGraph. may I ask how much a priority implementing line comments is? I know it's not the most useful feature, but not having to edit all my queries would save me a lot of time!

**Comment by jeffreylovitz** [comment]
Hi @odanoburu,

Line comments and block comments kind of just escaped our attention! We have an explicit whitelist of supported AST elements, and the comment formats haven't come up before.

I think this would be really trivial to correct, so will try to roll out a fix next week!

**Comment by odanoburu** [comment]
hello @jeffreylovitz , thank you for the quickfix! I've successfully (as far as I can tell) compiled your work at e3d613c1939d447e27f5f67d804830d202e01e3c, but when I try to load the module using redis I get:

```
% redis-server --loadmodule src/redisgraph.so 
53995:C 07 Sep 2020 14:50:20.171 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
53995:C 07 Sep 2020 14:50:20.171 # Redis version=5.0.9, bits=64, commit=00000000, modified=0, pid=53995, just started
53995:C 07 Sep 2020 14:50:20.171 # Configuration loaded
53995:M 07 Sep 2020 14:50:20.172 * Increased maximum number of open files to 10032 (it was originally set to 1024).
                _._                                                  
           _.-``__ ''-._                                             
      _.-``    `.  `_.  ''-._           Redis 5.0.9 (00000000/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._                                   
 (    '      ,       .-`  | `,    )     Running in standalone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 53995
  `-._    `-._  `-./  _.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |           http://redis.io        
  `-._    `-._`-.__.-'_.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |                                  
  `-._    `-._`-.__.-'_.-'    _.-'                                   
      `-._    `-.__.-'    _.-'                                       
          `-._        _.-'                                           
              `-.__.-'                                               

53995:M 07 Sep 2020 14:50:20.173 # Server initialized
53995:M 07 Sep 2020 14:50:20.173 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
53995:M 07 Sep 2020 14:50:20.173 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
53995:M 07 Sep 2020 14:50:20.173 # Module src/redisgraph.so failed to load: src/redisgraph.so: undefined symbol: output
53995:M 07 Sep 2020 14:50:20.173 # Can't load module from src/redisgraph.so: server aborting
```

do you have any idea what the problem could be? does it merit a new issue or it's a simple thing I'm not aware of?

**Comment by swilly22** [comment]
@odanoburu please try specifying a full absolute path to redisgraph.so

**Comment by odanoburu** [comment]
@swilly22 thank you for the tip! unfortunately it still didn't work :/

```
% redis-server --loadmodule /home/odanoburu/sites/redisgraph/src/redisgraph.so
66270:C 07 Sep 2020 15:32:49.582 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
66270:C 07 Sep 2020 15:32:49.582 # Redis version=5.0.9, bits=64, commit=00000000, modified=0, pid=66270, just started
66270:C 07 Sep 2020 15:32:49.582 # Configuration loaded
66270:M 07 Sep 2020 15:32:49.583 * Increased maximum number of open files to 10032 (it was originally set to 1024).
                _._                                                  
           _.-``__ ''-._                                             
      _.-``    `.  `_.  ''-._           Redis 5.0.9 (00000000/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._                                   
 (    '      ,       .-`  | `,    )     Running in standalone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 66270
  `-._    `-._  `-./  _.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |           http://redis.io        
  `-._    `-._`-.__.-'_.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |                                  
  `-._    `-._`-.__.-'_.-'    _.-'                                   
      `-._    `-.__.-'    _.-'                                       
          `-._        _.-'                                           
              `-.__.-'                                               

66270:M 07 Sep 2020 15:32:49.584 # Server initialized
66270:M 07 Sep 2020 15:32:49.584 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
66270:M 07 Sep 2020 15:32:49.585 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
66270:M 07 Sep 2020 15:32:49.586 # Module /home/odanoburu/sites/redisgraph/src/redisgraph.so failed to load: /home/odanoburu/sites/redisgraph/src/redisgraph.so: undefined symbol: output
66270:M 07 Sep 2020 15:32:49.586 # Can't load module from /home/odanoburu/sites/redisgraph/src/redisgraph.so: server aborting
```

one problem I had during compilation was with `peg/leg`; I'm using Fedora which does not have it packaged like Ubuntu, so I downloaded the source and compiled it; I only had to add a `-fPIC` flag to peg's Makefile (because libcypher-parser was complaining) and supply the appropriate paths as arguments, and it worked alright (or at least so it seems! I'm mentioning it in case it might be related.)

**Comment by odanoburu** [comment]
comparing the results of `nm` on my machine
```
% nm -C src/redisgraph.so | grep output
                 U output
0000000000562440 T tdefl_compress_mem_to_output
000000000055c1a0 t tdefl_output_buffer_putter
```

and on a container running the official image of redisgraph 2.2.5
```
# nm -C usr/lib/redis/modules/redisgraph.so | grep output
00000000005e9a80 T tdefl_compress_mem_to_output
00000000005e3b40 t tdefl_output_buffer_putter
```

so I guess there is some miscompilation?

**Comment by swilly22** [comment]
One thing that pops out of the Redis log is:
```
66270:M 07 Sep 2020 15:32:49.586 # Module /home/odanoburu/sites/redisgraph/src/redisgraph.so failed to load: /home/odanoburu/sites/redisgraph/src/redisgraph.so: undefined symbol: output
```

problem is `output` is too generic of a name to help me understand where it is being used.

**Comment by odanoburu** [comment]
yes, and interestingly this symbol does not exist in the previous binary, as shown by my previous comment.

I tried building the docker image using `make docker` because I thought there was no way this wouldn't work, but I got the same error, see this [gist](https://gist.github.com/odanoburu/de829dd98a97dd326fd7b73e686eb778) for all the output.

can you not reproduce this error? if so, is there I way I can get an up to date version some other way? (could you make a release and push it do docker hub, maybe?)

**Comment by swilly22** [comment]
I believe the `edge` tag should have that change
`docker run --rm -p 6379:6379 redislabs/redisgraph:edge`

**Comment by odanoburu** [comment]
hmm, does not seem like it! is it supposed to have the latest commit or the latest tag?

```
% docker run -p 6379:6379 -it --rm redislabs/redisgraph:edge
Getting image source signatures
Copying blob 73995e142582 skipped: already exists  
Copying blob 1b8430b65798 skipped: already exists  
Copying blob 6cdda0d59069 skipped: already exists  
Copying blob bf5952930446 skipped: already exists  
Copying blob 02ff1b34f4c8 skipped: already exists  
Copying blob 1e6a86949608 skipped: already exists  
Copying blob d50c7d79365d skipped: already exists  
Copying blob 28f3d4f425df skipped: already exists  
Copying blob b7f5beb0bf4b done  
Copying blob 5c5a17333434 done  
Copying config 218dc3bf77 done  
Writing manifest to image destination
Storing signatures
1:C 08 Sep 2020 20:10:31.895 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 08 Sep 2020 20:10:31.895 # Redis version=6.0.5, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 08 Sep 2020 20:10:31.895 # Configuration loaded
                _._                                                  
           _.-``__ ''-._                                             
      _.-``    `.  `_.  ''-._           Redis 6.0.5 (00000000/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._                                   
 (    '      ,       .-`  | `,    )     Running in standalone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 1
  `-._    `-._  `-./  _.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |           http://redis.io        
  `-._    `-._`-.__.-'_.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |                                  
  `-._    `-._`-.__.-'_.-'    _.-'                                   
      `-._    `-.__.-'    _.-'                                       
          `-._        _.-'                                           
              `-.__.-'                                               

1:M 08 Sep 2020 20:10:31.900 # Server initialized
1:M 08 Sep 2020 20:10:31.900 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
1:M 08 Sep 2020 20:10:31.900 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
1:M 08 Sep 2020 20:10:31.913 * <graph> Graph deletion will be done asynchronously.
1:M 08 Sep 2020 20:10:31.914 * <graph> Thread pool created, using 8 threads.
1:M 08 Sep 2020 20:10:31.914 * <graph> Maximum number of OpenMP threads set to 8
1:M 08 Sep 2020 20:10:31.914 * Module 'graph' loaded from /usr/lib/redis/modules/redisgraph.so
1:M 08 Sep 2020 20:10:31.914 * Ready to accept connections
```

```
% docker run -it --rm --entrypoint bash redislabs/redisgraph:edge
root@b21d77cafb36:/data# redis-cli -h 10.88.0.67  # ip of container running redis-server
10.88.0.67:6379> GRAPH.QUERY the-graph "/* opa */ RETURN true" --compact
(error) Encountered unsupported query type 'block_comment'
10.88.0.67:6379> GRAPH.QUERY the-graph "// ola\nRETURN true" --compact
(error) Encountered unsupported query type 'line_comment'
```

**Comment by swilly22** [comment]
Running against the edge image

```
127.0.0.1:6379> GRAPH.QUERY g "MATCH (n) return n /*comment*/"
1) 1) "n"
2) (empty array)
3) 1) "Cached execution: 0"
   2) "Query internal execution time: 1.414800 milliseconds"
```

Although I see the fix we've applied doesn't work as expected for some queries, like the one you've used.
I'm going to re-open this issue.

**Comment by odanoburu** [comment]
sorry @swilly22, just opened another issue with more details that you've probably already figured out by now!

**Comment by odanoburu** [comment]
I'm closing this since the original issue was solved here and in #1329. My compilation issues continue, but I can make do  with the Docker images.
